devicetypes/smartthings/zwave-radiator-thermostat.src/zwave-radiator-thermostat.groovy

```diff
@@ -0,0 +1,262 @@
+/**
+ *	Copyright 2019 SmartThings
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Z-Wave Radiator Thermostat", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.thermostat") {
+		capability "Thermostat Mode"
+		capability "Refresh"
+		capability "Battery"
+		capability "Thermostat Heating Setpoint"
+		capability "Health Check"
+		capability "Thermostat"
+		capability "Temperature Measurement"
+
+		command "setThermostatSetpointUp"
+		command "setThermostatSetpointDown"
+		command "switchMode"
+
+		fingerprint mfr: "0060", prod: "0015", model: "0001", deviceJoinName: "Everspring Thermostatic Radiator Valve", mnmn: "SmartThings", vid: "generic-radiator-thermostat"
+	}
+
+	tiles(scale: 2) {
```

Comment:
There's actually a better way to do the tiles now so that you don't need to do `switchMode` and `setThermostatSetpointUp` and `setThermostatSetpointDown`. Check out: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/f6f0b3ff7b31c83db0c78883f8383ca48ab04cb1/devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy#L42

Reply:
It works well, but when I try to change thermostat mode, I see only one (heat) on the list, despite all of them being sent as an event (and all of them are listed in IDE in supportedThermostatModes list).

Reply:
any idea on this @dkirker ? does the metadata only specify one mode, maybe?

Reply:
Make sure the payload for `supportedThermostatModes` is encoded as JSON.

Reply:
Yep, that've worked.